### PR TITLE
Submit transactions to Horizon

### LINKED ISSUE #23: Submit transactions

### PR DESCRIPTION
**Pull request by juanhurtado10**
Closes #23

### CODE REVIEW

File: lib/horizon/transactions.ex [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  @endpoint "/transactions"
+
+  @spec create(base64_envelope :: String.t()) :: response()
+  def create(base64_envelope) do
+    headers = [{"Content-Type", "application/x-www-form-urlencoded"}]
+    body = URI.encode_query(tx: base64_envelope)
+
+    with {:ok, tx} <- Horizon.request(:post, @endpoint, headers, body) do
+      {:ok, Transaction.new(tx)}
+    end
```
Comment (luishurtado) [inline lib/horizon/transactions.ex:26]: I don't get this `with` statement here. Why it is better than a `case`?. Also, what about the errors?

**Review by luishurtado — CHANGES_REQUESTED** [review]

File: lib/horizon/transactions.ex (hunk 1 above)
Comment (juanhurtado10) [reply lib/horizon/transactions.ex:26]: Errors are handled directly by the client implementation `Stellar.Horizon.Client.Default`, there are two reasons for this:
- Avoid handling errors on every module that interacts with Horizon. 
- Keep a standardized way to manage errors and remove code duplication.

The `with` statement is just a personal preference since errors are being managed by the client, no error clauses are needed using a `with`, if the condition is not met the not matching response from `Horizon.request/4` is returned.

Long story short .... it is equivalent to:

```elixir
    case Horizon.request(:post, @endpoint, headers, body) do
      {:ok, tx} -> Transaction.new(tx)
      error -> error
    end
```

Comment (juanhurtado10) [reply lib/horizon/transactions.ex:26]: Now ..... if you didn't get it probably the code is not declarative enough 😅 , let me change it!

**Review by luishurtado — APPROVED** [review]
